docs/walk-through/dags.md

```diff
@@ -0,0 +1,234 @@
+# DAGs
+
+DAGs (Directed Acyclic Graphs) are similar to `Steps`, and the syntax to use DAGs is exactly the same. DAGs are formed
```

Comment:
nit: I'd say "very similar" instead of "exactly the same" now you've not got "using context managers" in the sentence, because I'd consider the `>>` to be part of "the syntax" of DAGs.

Reply:
Fair point, changed some language and updated the paragraph to flow better.